src/go/k8s/internal/testutils/k3s.go

```diff
@@ -0,0 +1,605 @@
+	// based on the *original* DNS host prior to remapping the underlying
+	// address in our custom dialer.
+	tlsConfig.InsecureSkipVerify = true
+	tlsConfig.VerifyConnection = func(cs tls.ConnectionState) error {
+		opts := x509.VerifyOptions{
+			Roots:         config.RootCAs,
+			CurrentTime:   time.Now(),
+			DNSName:       serverName,
+			Intermediates: x509.NewCertPool(),
+		}
+
+		for _, cert := range cs.PeerCertificates[1:] {
+			opts.Intermediates.AddCert(cert)
+		}
+
+		if _, err := cs.PeerCertificates[0].Verify(opts); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	return tlsConfig
+}
+
+func (k *K3sStack) initializeK3s(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), k.Options.InstallationTimeout)
+	defer cancel()
+
+	container, err := k3s.Run(ctx, k.Options.K3sImage)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		container.Terminate(context.Background())
+	})
+
+	config, err := container.GetKubeConfig(ctx)
+	require.NoError(t, err)
+
+	k.config = config
+
+	restcfg, err := clientcmd.RESTConfigFromKubeConfig(config)
+	require.NoError(t, err)
+
+	k.rest = restcfg
+
+	s := runtime.NewScheme()
+	require.NoError(t, clientgoscheme.AddToScheme(s))
+	require.NoError(t, v1alpha2.AddToScheme(s))
+	require.NoError(t, clusterredpandacomv1alpha1.AddToScheme(s))
+
+	client, err := client.New(restcfg, client.Options{Scheme: s})
+	require.NoError(t, err)
+	k.client = client
+}
+
+func (k *K3sStack) initializeHelm(t *testing.T) {
+	dir, err := os.MkdirTemp("", "redpanda-test-helm")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		os.RemoveAll(dir)
+	})
+	k.directory = dir
+
+	options := &helmclient.Options{
+		RepositoryCache:  path.Join(dir, "cache"),
+		RepositoryConfig: path.Join(dir, "repo"),
+	}
+
+	kubeconfig := path.Join(dir, "kubeconfig")
+	err = os.WriteFile(kubeconfig, k.config, 0644)
+	require.NoError(t, err)
+	t.Logf("Kubeconfig: %s", kubeconfig)
+
+	client, err := helmclient.NewClientFromKubeConf(&helmclient.KubeConfClientOptions{
+		Options:    options,
+		KubeConfig: k.config,
+	})
+	require.NoError(t, err)
+	k.helm = client
+}
+
+func (k *K3sStack) installCertManagerChart(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), k.Options.InstallationTimeout)
+	defer cancel()
+
+	chartSpec := helmclient.ChartSpec{
+		ReleaseName: "cert-manager",
+		ChartName:   k.Options.CertManagerChart,
+		Namespace:   "cert-manager",
+		ValuesOptions: values.Options{
+			Values: []string{"crds.enabled=true"},
+		},
+		CreateNamespace: true,
+		Wait:            true,
+		Timeout:         k.Options.InstallationTimeout,
+	}
+
+	err := k.helm.AddOrUpdateChartRepo(repo.Entry{
+		Name: k.Options.CertManagerRepo,
+		URL:  k.Options.CertManagerURL,
+	})
+	require.NoError(t, err)
+
+	_, err = k.helm.InstallChart(ctx, &chartSpec, nil)
+	require.NoError(t, err)
+}
+
+func (k *K3sStack) installRedpandaChart(t *testing.T, options RedpandaOptions) {
```

Comment:
Would there be a reason to not use the flux reconciler to install the chart rather than adding a second method of installation?

Reply:
Ah, probably the need for cert manager...